hoot-core/src/main/cpp/hoot/core/cmd/DeriveChangesetCmd.cpp

```diff
@@ -47,50 +48,132 @@ class DeriveChangesetCmd : public BaseCommand
     }
 
-    LOG_INFO("Deriving changeset for inputs " << args[0] << ", " << args[1] << "...");
+    const QString input1 = args[0];
+    const QString input2 = args[1];
+    const QString output = args[2];
 
-    //use the same unknown1 status for both so they pass comparison correctly
-    OsmMapPtr map1(new OsmMap());
-    loadMap(map1, args[0], true, Status::Unknown1);
+    LOG_INFO(
+      "Deriving changeset for inputs " << input1 << ", " << input2 << " and writing output to " <<
+      output << "...");
 
+    //use the same unknown1 status for both so that difference doesn't influence the comparison
+    OsmMapPtr map1(new OsmMap());
+    loadMap(map1, input1, true, Status::Unknown1);
     OsmMapPtr map2(new OsmMap());
-    loadMap(map2, args[1], true, Status::Unknown1);
-
+    loadMap(map2, input2, true, Status::Unknown1);
     ElementSorterPtr sorted1(new ElementSorter(map1));
     ElementSorterPtr sorted2(new ElementSorter(map2));
     ChangesetDeriverPtr delta(new ChangesetDeriver(sorted1, sorted2));
 
-    if (args[2].endsWith(".osc"))
+    OsmChangesetXmlWriter xmlWriter;
+    if (isXmlOutput)
     {
-      OsmChangesetXmlWriter().write(args[2], delta);
+      xmlWriter.write(output, delta);
     }
-    else if (args[2].endsWith(".osc.sql"))
+    else
     {
-      if (args.size() != 4)
+      assert(!osmApiDbUrl.isEmpty());
+      LOG_DEBUG(osmApiDbUrl);
+      OsmChangesetSqlWriter(QUrl(osmApiDbUrl)).write(output, delta);
+    }
+
+    //There technically is no use case for recording the changeset file output in the db for sql
+    //changesets in addition to xml changesets, but the behavior is being made identical to that
+    //of xml changeset derivation for consistency's sake.
+
+    //write the output file name to the job status detail col for later retrieval
+    if (writeJobStatus)
+    {
+      assert(!hootApiDbUrl.isEmpty() && !jobId.isEmpty());
+      LOG_DEBUG(
+        "Associating changeset file: " << output << " with services job ID: " << jobId << "...");
+      LOG_VARD(hootApiDbUrl);
+      _hootApiDb.open(QUrl(hootApiDbUrl));
+      QString outputPath = output;
+      if (isXmlOutput && xmlWriter.getMultipleChangesetsWritten())
       {
-        throw HootException(
-          QString("SQL changeset writing requires a target database URL for configuration purposes."));
+        //The xml writer will break up changesets into multiple files if they exceed the maximum
+        //changeset size.  Thus, the path will be changed.  The sql writer does not break up the
+        //output into seperate files.
+
+        //We're just going to punt on multiple changeset files for now and make sure the services
+        //handle the situation appropriately.  We don't want to have to deal with the potentially
+        //large size of multiple changesets worth of data when returning changeset data from the
+        //services.
+        outputPath = "<multiple files>";
       }
```

Comment:
What does "handle the situation appropriately" mean for services?  Will it return multiple files?  Or does "punt" mean this situation will be unsupported?